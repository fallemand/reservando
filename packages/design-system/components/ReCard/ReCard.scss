@import '../../styles/variables';

.re-card {
  border-radius: $bdu;
  background-color: $white;
  padding: ($bdu * 2) ($bdu * 2.5);
  border: 1.5px solid $grey-400;
}
