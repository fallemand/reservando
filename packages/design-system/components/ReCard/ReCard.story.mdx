import { Meta, Story, ArgsTable, Canvas, Description } from '@storybook/addon-docs/blocks';
import { ReCard } from "@reservando/design-system";

<Meta title="Components/Card" component={ReCard} />

# ReCard

<Canvas>
  <Story name="basic">
    {{
      components: { ReCard },
      data() {
        return {
          checked: true,
        };
      },
      template: `
        <ReCard>
          My card content
        </ReCard>
      `,
    }}
  </Story>
</Canvas>

<Description of={ReCard} />

## How to use

```html
<script lang="ts">
  import { ReCard } from '@reservando/design-system';
</script>
```

<ArgsTable of={ReCard} />
