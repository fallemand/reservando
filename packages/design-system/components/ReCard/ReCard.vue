<template>
  <div class="re-card" :class="$attrs.class">
    <slot />
  </div>
</template>

<script lang="ts">
import { defineComponent } from "vue";

const ReCard = defineComponent({
  name: "ReCard",
});

export default ReCard;
</script>

<style lang="scss" src="./ReCard.scss" />
