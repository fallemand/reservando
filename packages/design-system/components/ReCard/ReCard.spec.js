import { shallowMount } from "@vue/test-utils";
import ReCard from "./ReCard.vue";

describe("ReCard component", () => {
  const slots = {
    default: "__CONTENT__",
  };

  it("should render with default props", () => {
    const wrapper = shallowMount(ReCard, { slots });
    expect(wrapper.element).toMatchSnapshot();
  });
});
