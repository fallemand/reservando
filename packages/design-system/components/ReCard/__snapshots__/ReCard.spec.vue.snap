// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReCard component should render with default props 1`] = `
<div
  class="re-card"
>
  
  __CONTENT__
  
</div>
`;
